### ISSUE TITLE
OSX/Mac Verision

### ISSUE THREAD

**Issue by bluecursor** [excess-c64/v1541commander#1]
Looking for a OSX/Mac Version. Possible?

**Comment by Zirias** [comment]
Possible indeed, all the code is fairly portable with only a few "special paths" for Windows, assuming all other systems are more or less POSIX compliant, and as far as I know, MacOS X fullfills this. Qt (which is used for the GUI) is available for MacOS X.

Still I can't do it. I don't own a Mac, and Apple tries hard to force anyone wanting to build software for MacOS X to actually buy and use a Mac ...

Maybe some Apple user with programming and/or software building experience could help here. If you know someone, please establish contact with me, thanks :)

Until that (magically?) happens, try Wine. In my experiments on FreeBSD, V1541Commander for win32 works correctly in Wine, so I assume it does as well on MacOS X.

**Comment by bluecursor** [comment]
I tried with Wine app Called Crossover and it doesn't work.

**Comment by Zirias** [comment]
Do you by any chance run "Catalina"? If so, there isn't a solution at the moment:
https://forum.winehq.org/viewtopic.php?t=32590

With older versions of MacOS X, it should work. Just double checked with my 32bit wine 3.0.4 on FreeBSD, no problem.

Of course, a native MacOS X package would be great, and as I said, it should be possible, but I can't create one without owning a Mac :( I'll definitely leave this issue open. Maybe someone can help here!

**Comment by bluecursor** [comment]
That problem has been cracked:
https://www.codeweavers.com/about/blogs/jwhite/2019/10/31/announcing-a-first-alpha-build-of-crossover-19-for-macos-catalina

Setup.exe and uninstall.exe work file.
Main App comes back with a "returned 1" error

**Comment by Zirias** [comment]
Well, "Crossover" shares (a lot of) code with wine, but isn't the same. And I read up on their solution for 32bit apps on Catalina, which seems to be a remarkable hack (so, great work), but I wouldn't count on it to get everything correct. Maybe the issue is just the wine version -- unfortunately, there is no newer 32bit wine version than the pretty old 3.0.4 available for FreeBSD yet, so I can't test that.

A possible workaround could be a 64bit Windows build, which might run in wine on Catalina. I don't have a cross-build environment for win64 right now, but could setup one (but it takes time). Still, even if this works, it's not a great solution -- and for Windows, a 64bit build has no advantage at all over the existing 32bit build.

In short: The only sensible way forward here is a native build of V1541Commander for OS X. As I said, I don't see how I can create such a build right now, and hope for help coming ;)

**Comment by koimark** [comment]
I wish I could have skills to made OS X build. Maybe I will try someday.

**Comment by cwscws** [comment]
Any news on a Mac version?

**Comment by Zirias** [comment]
Kind of. As already explained, I can't do it myself, but I got help now. There were a few code changes and it's now confirmed to build and run on Mac, but still needs a lot of hackery in the build system to achieve that, which turned out to be the bigger issue as MacOS has its very own mechanism for shared libraries .... here, many thanks to @Koston-0xDEADBEEF for help, testing, and suggesting changes!

I will need to find some time to review and properly integrate the changes to the build system needed for MacOS. Then, it could at least be used in e.g. MacPorts.

**Comment by Koston-0xDEADBEEF** [comment]
[v1541commander_macos13.tar.gz](https://github.com/excess-c64/v1541commander/files/12005921/v1541commander_macos13.tar.gz)

Here's binary of v1541commander and accompanying library - for macOS/arm64. Needs `qt5-qtbase` package from MacPorts.

**Comment by Zirias** [comment]
Thanks again @Koston-0xDEADBEEF, awesome! I really need the time to review/integrate your `zimk` changes, so ppl could at least easily build it for any architecture.

**Comment by cwscws** [comment]
Great news! Would be nice to make a .app for easy usage…

**Comment by Koston-0xDEADBEEF** [comment]
> Great news! Would be nice to make a .app for easy usage…

After @Zirias is done restructing the `zimk` build system to support macOS, the `v1541commander` will be ported to [MacPorts](https://github.com/macports/macports-ports/). Once that is done, various macOS-native binary distribution formats can be created with MacPorts builtin functionality.

**Comment by cwscws** [comment]
Best news today! 😀

**Comment by cwscws** [comment]
Any news on this issue?

**Comment by Koston-0xDEADBEEF** [comment]
Zirias still needs to restructure his `zimk` build system to support macOS, but I think that's not going to happen anytime soon (if at all) because it requires very intrusive changes due to the way Apple has changed semantics of building and using dynamic libraries on macOS. Zirias would need to acquire access to a macOS system and then more or less rewrite major parts of `zimk` from scratch.
